cpp/dynamic_community_detection_module/algorithm/dynamic_community_detection.hpp

```diff
@@ -0,0 +1,241 @@
+
+namespace LabelRankT {
+
+class LabelRankT {
+ private:
+  /* #region parameters */
+  /// default edge weight
+  double DEFAULT_WEIGHT = 1;
+
+  /// graph directedness
+  bool is_directed = false;
+  /// whether to use weights
+  bool is_weighted = false;
+
+  /// similarity threshold used in the node selection step, values within [0, 1]
+  double similarity_threshold = 0.7;
+  /// exponent used in the inflation step
+  double exponent = 4;
+  /// lowest probability at the cutoff step
+  double min_value = 0.1;
+
+  /// weight property name
+  std::string weight_property = "weight";
+  /// default self-loop weight
+  double w_selfloop = 1.0;
+
+  /// maximum number of iterations
+  std::uint64_t max_iterations = 100;
+  /// maximum number of updates for any node
+  std::uint64_t max_updates = 5;
+  /* #endregion */
+
+  /* #region structures */
+  /// reference to current graph
+  std::unique_ptr<mg_graph::Graph<>> graph;
+
+  /// map containing each node’s community label probabilities
+  std::unordered_map<std::uint64_t, std::unordered_map<std::uint64_t, double>>
+      label_Ps;
+  /// sum of weights of each node’s in-edges
+  std::unordered_map<std::uint64_t, double> sum_w;
+  /// how many times each node has been updated
+  std::unordered_map<std::uint64_t, std::uint64_t> times_updated;
+
+  /// flag whether community labels have already been calculated
+  bool calculated = false;
+  /* #endregion */
+
+  /* #region helper_methods */
+  ///@brief Returns a vector of all graph nodes’ Memgraph IDs.
+  ///
+  ///@return -- vector of all graph nodes’ Memgraph IDs
+  std::vector<std::uint64_t> NodesMemgraphIDs();
+
+  ///@brief Returns in-neighbors’ Memgraph IDs for a given node’s Memgraph ID.
+  /// If the graph is undirected, all neighbors are included.
+  ///
+  ///@param node_id -- given node’s Memgraph ID
+  ///
+  ///@return -- vector of neighbors’ Memgraph IDs
+  std::unordered_set<std::uint64_t> InNeighborsMemgraphIDs(
+      const std::uint64_t node_id);
+
+  ///@brief Initializes internal data structures.
+  void SetStructures(const std::uint64_t node_id);
+
+  ///@brief Removes deleted nodes’ entries from internal data structures.
+  ///
+  ///@param deleted_nodes_ids -- set of deleted nodes’ IDs
+  void RemoveDeletedNodes(const std::unordered_set<std::uint64_t>& deleted_nodes_ids);
+
+  ///@brief Retrieves the total weight of edges between two nodes.
+  /// With undirected graphs it considers all edges between the given node pair.
+  /// On a directed graph, it checks if the source and target nodes match.
+  ///
+  ///@param from_node_id -- source node’s Memgraph ID
+  ///@param to_node_id -- target node’s Memgraph ID
+  ///
+  ///@return -- total weight of edges between from_node_id and to_node_id
+  double GetTotalWeightBetween(const std::uint64_t from_node_id,
+                               const std::uint64_t to_node_id);
+
+  ///@brief Retrieves given edge’s weight.
+  double GetWeight(const std::uint64_t edge_id);
```

Comment:
```suggestion
  double GetWeight(const std::uint64_t edge_id) const;
```

Reply:
consider const qualifying the rest of the member functions that don't modify data members.

Reply:
Done!